modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RangeCondition.java

```diff
@@ -29,9 +31,15 @@ public interface RangeCondition<RowT> {
     /** Upper search row. */
     RowT upper();
 
-    /** Inlusive search by lower row. */
+    /** Return positions of unspecified (i.e. null equal) nodes depends to lower bound. */
+    ImmutableBitSet unspecifiedLower();
```

Comment:
ticket description states `After reworking expandBounds the searchRow that comes to RowConverter#toBinaryTuplePrefix should already contain a prefix only.`. With such an approach, we still have nulls at the end of a searchRow.

I would suggest to rework this patch, so both `lower` and 'upper' will return only real prefixes of index keys. Here is an example: if table row is (a, b, c), we have index (c, a) and predicate `WHERE c > 0`, then `lower` should be `Row [0]` (currently it is `Row [0, null]`)